Source/MQTTnet/Implementations/MqttTcpServerAdapter.cs

```diff
@@ -44,13 +44,17 @@ public Task StartAsync(IMqttServerOptions options)
 
             if (options.TlsEndpointOptions?.IsEnabled == true)
             {
-                if (options.TlsEndpointOptions.Certificate == null)
+                if (options.TlsEndpointOptions.Certificate == null && options.TlsEndpointOptions.X509Certificate == null)
```

Comment:
Please throw an exception if both is set. In such cases we cannot decide which certificate must be used.